### PR TITLE
Update release note for Spark 3.5.3

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes to update release note for Spark 3.5.3

Some tickets are not actually contained from 3.5.3, so I removed them from list and also updated the target version from associated JIRA.

### CODE REVIEW

File: site/docs/latest [hunk 1]
```diff
@@ -1 +1 @@
-3.5.2
\ No newline at end of file
+3.5.3
```
Comment (dongjoon-hyun) [inline site/docs/latest]: Oh, is this different from?
- #559

Comment (itholic) [reply site/docs/latest]: No, actually this is not related to this PR. I wonder why this changed is included here

Comment (itholic) [reply site/docs/latest]: Interesting... I didn't touch anything and even it's already points out 3.5.3 from latest `asf-site` lol

Comment (itholic) [reply site/docs/latest]: Ah, it was not properly rebased to master(`asf-site`). Just resolved it :-)

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

cc @cloud-fan , @HyukjinKwon

**Comment by dongjoon-hyun** [comment]
Oh, please fix the CIs, @itholic .

**Review by HyukjinKwon — APPROVED** [review]

**Comment by itholic** [comment]
Merged to asf-site. Thanks @dongjoon-hyun and @HyukjinKwon !

File: releases/_posts/2024-09-24-spark-release-3-5-3.md [hunk 2]
```diff
@@ -22,7 +22,6 @@ Spark 3.5.3 is the third maintenance release containing security and correctness
 - [[SPARK-46037]](https://issues.apache.org/jira/browse/SPARK-46037) - When Left Join build Left, ShuffledHashJoinExec may result in incorrect results
 - [[SPARK-46632]](https://issues.apache.org/jira/browse/SPARK-46632) - EquivalentExpressions throw IllegalStateException
 - [[SPARK-48965]](https://issues.apache.org/jira/browse/SPARK-48965) - toJSON produces wrong values if DecimalType information is lost in as[Product]
-- [[SPARK-49030]](https://issues.apache.org/jira/browse/SPARK-49030) - Self join of a CTE seems non-deterministic
```
Comment (cloud-fan) [inline releases/_posts/2024-09-24-spark-release-3-5-3.md:25]: does it mean these tickets miss specify the fixed version?

Comment (itholic) [reply releases/_posts/2024-09-24-spark-release-3-5-3.md:25]: Yes, they are not included (or even not resolved) in 3.5.3, so I also updated "Fix Version/s" to 3.5.4:

Screenshot 2024-09-25 at 4 48 45 PM

**Pull request closed without merging**